File: develop/lab-3-dynamodb/dynamoDBJavaLab/src/test/java/com/amazonaws/lab/CustomerReportLinkerTest.java
package com.amazonaws.lab;
// Copyright 2017 Amazon Web Services, Inc. or its affiliates. All rights reserved.

import static org.junit.Assert.*;

import org.junit.Test;

public class CustomerReportLinkerTest {

	@Test
	public void test() throws Exception {
		CustomerReportLinker.main(new String[0]);
		for (int i = 1; i < 4; i++) {
			// Checks if CustomerReportUrl attribute is present.
			boolean customerReportUrlAttrPresent = CustomerReportLinker.dynamoDB
					.getTable(CustomerReportLinker.RESERVATIONS_TABLE_NAME).getItem("CustomerId", ("" + i))
					.isPresent("CustomerReportUrl");
			System.out.printf("Test - CustomerId: %s, customerReportUrlAttrPresent: %b %n", i,
					customerReportUrlAttrPresent);
			assertTrue(customerReportUrlAttrPresent);
		}
	}
}

File: develop/lab-3-dynamodb/dynamoDBPythonLab/customer_report_linker_test.py
# Copyright 2017 Amazon Web Services, Inc. or its affiliates. All rights
# reserved.

import unittest
import boto3
import warnings
import utils

TABLE_NAME = utils.LAB_S3_RESERVATIONS_TABLE_NAME


class CustomerReportLinkerTest(unittest.TestCase):

    def test_report_link(self):
        warnings.simplefilter("ignore", ResourceWarning)
        dynamodb = boto3.resource('dynamodb')
        myTable = dynamodb.Table(TABLE_NAME)
        for i in range(1, 4):
            isPresent = False
            rec = myTable.get_item(
                Key={'CustomerId': str(i)})
            if rec.get('Item').get('CustomerReportUrl'):
                isPresent = True
            self.assertTrue(isPresent)

if __name__ == '__main__':
    unittest.main()
